[TypeScript]: autocomplete is not available on ContextValue<T, E>

### Provide a general summary of the issue here

This is because it is written as:

```tsx
type SlottedContextValue<T> = SlottedValue<T> | T | null | undefined;
export type ContextValue<T, E> = SlottedContextValue<WithRef<T, E>>;
```

Looking at the type, my guess is `SlottedContextValue` should be:

```ts
type SlottedContextValue<T> = SlottedValue<T> & T | null | undefined;
```

May be a mistake?
I can open a PR for it.

### 🤔 Expected Behavior?

Autocompletion should be available to the value of type `ContextValue<T, E>`.

```ts
type MyProps = { a: string }

type MyContextValue = ContextValue<MyCompProps, HTMLDivElement>

const MyContext = createContext<MyContextValue>(null)


function Comp() {
  return (
    <MyContext.Consumer>
      {(context) => {
         // autocompletion is not available
         console.log(context.a)
         return <div>{context}</div>
      }}
    </MyContext.Consumer>
  )
}
```

### 😯 Current Behavior

see above

### 💁 Possible Solution

see above

### 🔦 Context

_No response_

### 🖥️ Steps to Reproduce

See above

### Version

1.14.0

### What browsers are you seeing the problem on?

Other

### If other, please specify.

code

### What operating system are you using?

macos

### 🧢 Your Company/Team

Palo Alto Networks

### 🕷 Tracking Issue

_No response_

btw, the `CONTRIBUTING.md` instruction is still lacking.

Following it to run `yarn install` and `yarn start` (and `yarn build`), `yarn jest`/`yarn test` still results in failed tests such as:

`Cannot find module './Button.main.js' from 'packages/@react-spectrum/button/dist/main.js'`

> btw, the `CONTRIBUTING.md` instruction is still lacking.
> 
> Following it to run `yarn install` and `yarn start` (and `yarn build`), `yarn jest`/`yarn test` still results in failed tests such as:
> 
> `Cannot find module './Button.main.js' from 'packages/@react-spectrum/button/dist/main.js'`
> 

Can you open a separate issue with this? Please provide node version, os, etc. I wasn't seeing this before our break but I can pull a new copy and try again in the new year. The more info the better though for when one of us gets to it.

Also, it's running on CI just fine with those commands.

--- a/packages/@react-spectrum/s2/src/CoachMark.tsx
+++ b/packages/@react-spectrum/s2/src/CoachMark.tsx
@@ -320,7 +320,7 @@ const actionButtonSize = {
   XL: 'L'
 } as const;
 
-export const CoachMarkContext = createContext<ContextValue<CoachMarkProps, HTMLElement>>({});
+export const CoachMarkContext = createContext<ContextValue<Partial<CoachMarkProps>, HTMLElement>>({});
 
 export const CoachMark = forwardRef((props: CoachMarkProps, ref: ForwardedRef<HTMLElement>) => {
   let colorScheme = useContext(ColorSchemeContext);

--- a/packages/react-aria-components/src/utils.tsx
+++ b/packages/react-aria-components/src/utils.tsx
@@ -20,7 +20,7 @@ interface SlottedValue<T> {
   slots?: Record<string | symbol, T>
 }
 
-export type SlottedContextValue<T> = SlottedValue<T> | T | null | undefined;
+export type SlottedContextValue<T> = (SlottedValue<T> & T) | null | undefined;
 export type ContextValue<T, E> = SlottedContextValue<WithRef<T, E>>;
 
 type ProviderValue<T> = [Context<T>, T];

